feat: Add TF samples for creating a Looker (Google Cloud core) Enterprise instance that uses PSC

ronnieplooker opened the pull request:
## Description

Fixes b/429215192

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [x] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [x] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2023 Google LLC
```
Since this is a new sample, it should have the current date
```suggestion
 * Copyright 2025 Google LLC
```

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,35 @@
+ */
+# [START looker_google_looker_instance_psc]
+# Create an ENTERPRISE edition Looker (Google Cloud core) instance that has PSC enabled.
+resource "google_looker_instance" "main" {
```
If there's only one resource of any type, it should be called "default"

```suggestion
resource "google_looker_instance" "default" {
```

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,35 @@
+    client_secret = "my-client-secret"
+  }
+  psc_config {
+    allowed_vpcs = ["projects/test-project/global/networks/test"]
```
This value won't work, as `test-project` is not a valid project ID.

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,35 @@
+  }
+  psc_config {
+    allowed_vpcs = ["projects/test-project/global/networks/test"]
+    # update only
+    # service_attachments = [{local_fqdn: "www.local-fqdn.com" target_service_attachment_uri: "projects/my-project/regions/us-east1/serviceAttachments/sa"}]
```
Are these comments needed?

glasnt reviewed (commented):
Generally this sample looks okay, but other samples in the looker folder have configurations to skip tests. It will probably be needed for this one. 

Example: https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/looker/looker_instance_basic/test.yaml, where `metadata.name` is the folder name and will need to be changed.

ronnieplooker commented:
@glasnt thanks for taking a look! I have hopefully addressed your comments. If it helps, I took the sample directly from what the SME engineer uploaded to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/looker_instance#example-usage---looker-instance-psc.

glasnt commented on looker/looker-core-psc/test.yaml:
```diff
@@ -0,0 +1,20 @@
+apiVersion: blueprints.cloud.google.com/v1alpha1
+kind: BlueprintTest
+metadata:
+  name: looker_instance_psc
```
```suggestion
  name: looker_core_psc
```

glasnt commented:
/gcbrun

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,36 @@
+  psc_config {
+    allowed_vpcs = ["projects/test-project/global/networks/test"]
+    # (Optional) List of VPCs that are allowed ingress into the Looker instance.
+    service_attachments = [{local_fqdn: "www.local-fqdn.com" target_service_attachment_uri: "projects/my-project/regions/us-east1/serviceAttachments/sa"}]
```
If I'm reading the docs correctly, you don't set these values, they are updated for you. Trying to validate this part of the Terraform is failing with `An argument named "service_attachments" is not expected here.`. Are you able to confirm this is intended as an argument?

ronnieplooker replied:
Ah, I get it! So the service attachment argument is relevant only when updating an instance config, not when creating it. I think that to match the other blocks in https://cloud.google.com/looker/docs/looker-core-create-psc#create_instance we should remove that argument entirely.

As for the `allowed_vpcs`, you would only specify that if you are creating an instance that uses only private IP (that is:

```
private_ip_enabled = true
public_ip_enabled  = false
psc_enabled  = true
```
This example doesn't follow that pattern, so I have commented out the `allowed_vpcs` and added some context.

glasnt commented:
/gcbrun

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,34 @@
+  psc_config {
+    # allowed_vpcs = ["projects/{project}/global/networks/{network}"]
+    # (Optional) List of VPCs that are allowed ingress into the Looker instance. Set an allowed VPC if you are creating an instance that uses only private IP.
+   }
```
```suggestion
  }
```

glasnt commented:
/gcbrun

glasnt commented on looker/looker-core-psc/main.tf:
```diff
@@ -0,0 +1,34 @@
+ /**
```
```suggestion
/**
```
